src/Microsoft.Diagnostics.Monitoring.EventPipe/DiagnosticsEventPipeProcessor.cs

```diff
@@ -51,6 +57,11 @@ public async Task Process(DiagnosticsClient client, TimeSpan duration, Cancellat
 
                     Stream sessionStream = await streamProvider.ProcessEvents(client, duration, token);
 
+                    if (!_sessionStarted.TrySetResult(true))
```

Comment:
Not for this PR, but I'm wondering if this would be useful as part of the pipe api itself, or is this really just an issue in testing. In the future, if we ever make the pipeline api public, consumers may run into the exact same issue.

Reply:
Thinking about this more, when we do triggers, we might want to be able to call Pipeline.Start rather than .Run in order to start the pipelines and not necessarily care when they finish.